File: NodeJS/data/save_json.js
// Save full page to JSON file

const { chromium } = require('playwright');
const fs = require('fs');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  await page.goto('https://example.com');
  const html = await page.content();
  fs.writeFileSync('page.json', JSON.stringify({ html }, null, 2));
  await browser.close();
})();

File: Python/save data/save_json.py
# Save full page to JSON file

from playwright.sync_api import sync_playwright
import json

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page()
    page.goto("https://example.com")
    content = page.content()
    with open("page.json", "w", encoding="utf-8") as f:
        json.dump({"html": content}, f, ensure_ascii=False, indent=2)
    browser.close()
